yellow-paper/docs/addresses-and-keys/specification.md

```diff
@@ -559,103 +428,44 @@ This tag can be used as the basis for note retreival schemes. Each time Bob send
-$\hmencheader$ | h("?", $\sharedsecret_{m,header}$) | Incoming encryption key |
-$\ciphertextheader$ | $enc_{\hmencheader}^{\Ivpkm}$(app_address)
-$\payload$ | [$\Taghs$, $\Epk_{hs}$, $\Epk$, $\ciphertextheader$] | Payload | This can be broadcast via L1.<br />Curve points can be compressed in the payload. |
-
-Alice can identify she is the indended the handshake recipient as follows:
-
-<!-- prettier-ignore -->
-| Thing | Derivation | Name | Comments |
-|---|---|---|---|
-$\Taghs$ | $\tskm \cdot \Epk_{hs}$ | Handshake message identification tag | Alice can extract $\Taghs$ and $\Epk_{hs}$ from the $\payload$ and perform this scalar multiplication on _every_ handshake message. If the computed $\Taghs$ value matches that of the $\payload$, then the message is indented for Alice.<br />Clearly, handshake transactions will need to be identifiable as such (to save Alice time), e.g. by revealing the contract address of some canonical handshaking contract alongside the $\payload$.<br />Recall: this step is merely an optimization, to enable Alice to do a single scalar multiplication before moving on (in cases where she is not the intended recipient). |
-
-If Alice successfully identifies that she is the indended the handshake recipient, she can proceed with deriving the shared secret (for tagging) as follows:
-
-<!-- prettier-ignore -->
-| Thing | Derivation | Name | Comments |
-|---|---|---|---|
-$\sharedsecret_{m,header}$ | $\ivskm \cdot \Epk$ | Shared secret, for encrypting the ciphertext header |
-$\hmencheader$ | h("?", $\sharedsecret_{m,header}$) | Incoming encryption key |
-app_address | $decrypt_{\hmencheader}^{\ivskm}(\ciphertextheader)$ |
-$\ivskapp$ | See derivations above. Use the decrypted app_address. | app-specific  incoming viewing secret key |
-$\sharedsecret_{app,tagging}^{Bob \rightarrow Alice}$ | $\ivskapp \cdot \Epk_{hs}$ | Shared secret, for tagging |  |
-
-A sequence of tags can then be derived by both Alice and Bob as:
-
-<!-- prettier-ignore -->
-| Thing | Derivation | Name | Comments |
-|---|---|---|---|
-$\tagg_{app,i}^{Bob \rightarrow Alice}$ | $h(\sharedsecret_{app,tagging}^{Bob \rightarrow Alice}, i)$ | The i-th tag in the sequence. |  |
-
-This tag can be used as the basis for note retreival schemes. Each time Bob sends Alice a $\ciphertext$ **for this particular app**, he can attach the next unused $\tagg_{app,i}^{Bob \rightarrow Alice}$ in the sequence. Alice - who is also able to derive the next $\tagg_{app,i}^{Bob \rightarrow Alice}$ in the sequence - can make privacy-preserving calls to a server, requesting the $\ciphertext$ associated with a particular $\tagg_{app,i}^{Bob \rightarrow Alice}$.
-
-> TODO: Bob can encrypt a record of this handshake, for himself, using his outgoing viewing key.
-
```

Comment:
I guess this chunk has been removed because it was just an illustration of one note discovery method?
It might be a useful reference to keep somewhere (not here), for when we build an opinionated note discovery scheme into an example wallet?

Reply:
It was removed because it relied on a per-app siloed incoming viewing key, which we had dropped since we couldn't find an efficient scheme to implement it. The section on "deriving a sequence of tags _across all apps_" is still here!

Reply:
Ah, got you! Thanks!